### PR TITLE
extended the users + organizations queries

### PR DESCRIPTION
**Pull request by techsmyth**
with optional parameters to allow limiting the number of users or orgs returned and what the selection criteria should be random or not

### CODE REVIEW

File: src/domain/community/organization/organization.resolver.queries.ts [hunk 1]
```diff
@@ -13,8 +13,28 @@ export class OrganizationResolverQueries {
...
+    @Args({
+      name: 'limit',
+      type: () => Float,
+      description:
+        'The number of Organizations to return; if omitted return all Organizations.',
+      nullable: true,
+    })
+    limit: number,
+    @Args({
+      name: 'randomSelection',
```
Comment (hero101) [inline src/domain/community/organization/organization.resolver.queries.ts]: `shuffle` would be more appropriate. I would avoid complex, long parameter names like that

File: src/domain/community/organization/organization.service.ts [hunk 2]
```diff
@@ -307,9 +308,28 @@ export class OrganizationService {
...
-    const organizations = await this.organizationRepository.find();
-    return organizations || [];
+  async getOrganizations(
+    limit?: number,
+    randomSelection = false
+  ): Promise<IOrganization[]> {
+    const organizations: IOrganization[] =
+      await this.organizationRepository.find();
+    if (!organizations) return [];
+    if (!limit || organizations.length < limit) return organizations;
```
Comment (hero101) [inline src/domain/community/organization/organization.service.ts]: you are not taking in account the `randomSelection` param

File: src/domain/community/organization/organization.service.ts [hunk 3]
```diff
@@ -307,9 +308,28 @@ export class OrganizationService {
...
+
+    // Need to restrict the set of users to return
+    if (randomSelection) {
+      const randomIndexes = generateRandomArraySelection(
+        limit,
```
Comment (hero101) [inline src/domain/community/organization/organization.service.ts]: in order to get the code executed through here, you need to pass the `limit` param.

File: src/domain/community/user/user.service.ts [hunk 4]
```diff
@@ -475,8 +476,21 @@ export class UserService {
     return user;
   }
 
-  async getUsers(): Promise<IUser[]> {
-    return (await this.userRepository.find({ serviceProfile: false })) || [];
+  async getUsers(limit?: number, randomSelection = false): Promise<IUser[]> {
+    const result = await this.userRepository.find({ serviceProfile: false });
+    if (!result) return [];
+    if (!limit || result.length < limit) return result;
```
Comment (hero101) [inline src/domain/community/user/user.service.ts]: not taking in account the `randomSelection` param again

**Review by hero101 — CHANGES_REQUESTED** [review]

File: src/domain/community/organization/organization.resolver.queries.ts (hunk 1 above)
Comment (techsmyth) [reply src/domain/community/organization/organization.resolver.queries.ts]: addressed in9a57bcd8860c2ff4031bed6d103313c4b6c9ba8f

File: src/domain/community/organization/organization.service.ts (hunk 2 above)
Comment (techsmyth) [reply src/domain/community/organization/organization.service.ts]: addressed in 9a57bcd8860c2ff4031bed6d103313c4b6c9ba8f

Comment (techsmyth) [reply src/domain/community/organization/organization.service.ts]: Note: shuffle paramter is only applied in the case where a limit is set.

File: src/domain/community/organization/organization.service.ts (hunk 3 above)
Comment (techsmyth) [reply src/domain/community/organization/organization.service.ts]: see above

File: src/domain/community/user/user.service.ts (hunk 4 above)
Comment (techsmyth) [reply src/domain/community/user/user.service.ts]: see above

**Review by hero101 — DISMISSED** [review]

**Review by hero101 — APPROVED** [review]

**Pull request merged**